shop/src/main/java/ro/msg/learning/shop/configuration/StockConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package ro.msg.learning.shop.configuration;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.repository.IRepositoryLocation;
+import ro.msg.learning.shop.repository.IRepositoryStock;
+import ro.msg.learning.shop.strategy.IDeliveryStrategy;
+import ro.msg.learning.shop.strategy.MostAbundantStrategy;
+import ro.msg.learning.shop.strategy.SingleLocationStrategy;
+
+@Configuration
+@AllArgsConstructor
+public class StockConfiguration {
+
+    private final IRepositoryStock repositoryStock;
+    private final IRepositoryLocation repositoryLocation;
+
+    @Bean
+    public IDeliveryStrategy selectStrategy(@Value("${selectedStrategy}") String strategy){
+        if(strategy.equals("mostAbundant")){
```

Comment:
You can create Enumeration for StrategyType and remove hardcoded text.

Reply:
Fixed.